src/kOS.Safe/Execution/Stack.cs

```diff
@@ -203,22 +215,16 @@ public List<int> GetCallTrace()
         }
         
         /// <summary>
-        /// Check if the call-stack has evidence that we are currently inside
-        /// a trigger.
+        /// This stack tracks all its pushes and pops to know whether or not it
+        /// contains subroutine contexts which are from triggers.  If there are
+        /// any still triggers in the stack, this returns true, else false.  
         /// </summary>
         /// <returns>True if the current call stack indicates that either we are
         /// inside a trigger, or are inside code that was in turn indirectly called
         /// from a trigger.  False if we are in mainline code instead.</returns>
         public bool HasTriggerContexts()
         {
-            for (int index = stackPointer + 1; index < stack.Count; ++index)
-            {
-                SubroutineContext contextRecord = stack[index] as SubroutineContext;
-                if (contextRecord != null)
-                    if (contextRecord.IsTrigger)
-                        return true;
-            }
-            return false;
+            return triggerContextCount > 0;
```

Comment:
The new triggerContextCount is what allowed the expense of this walk to be ripped out, which in turn allows us to just call ``stack.HasTriggerContext()`` willy-nilly whenever we feel like without it slowing things down.